test/unit/select_plan_test.lua

```diff
@@ -339,3 +339,201 @@ g.test_table_count = function()
     t.assert_equals(utils.table_count(
         {['Ivan'] = 1, ['Peter'] = 2, ['Fyodor'] = 3, ['Alexander'] = 4}), 4)
 end
+
+g.test_extract_sharding_key_from_condtitions = function()
+    local space_indexes = box.space.customers.index
+    local space_format = box.space.customers:format()
+    local fieldno_map = utils.get_format_fieldno_map(space_format)
+
+    -- Sharded by id (primary index).
+    local sharding_index = { parts = {{ fieldno = 1 }} }
+    local conditions = compare_conditions.parse({{ '==', 'id', 2 }})
+    local sharding_key = select_plan._internal.extract_sharding_key_from_conditions(conditions, sharding_index,
+                                                                                    space_indexes, fieldno_map)
+    t.assert_equals(sharding_key, { 2 }, "Extracted sharding key from equal condition for pk index")
```

Comment:
If this assert() will be broken then all cases below will be blocked by it.
It is better to parameterize testcase like you did it in `test/integration/ddl_sharding_key_test.lua`.

Reply:
Fair, separated test cases